homeassistant/components/sensor/zestimate.py

```diff
@@ -0,0 +1,173 @@
+"""
+Support for zestimate data from zillow.com.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.zestimate/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+import requests
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY,
+                                 CONF_NAME, STATE_UNKNOWN, ATTR_ATTRIBUTION)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['xmltodict==0.11.0']
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'http://www.zillow.com/webservice/GetZestimate.htm'
+
+CONF_ZPID = 'zpid'
+CONF_ATTRIBUTION = "Data provided by Zillow.com"
+
+DEFAULT_NAME = 'Zestimate'
+NAME = 'zestimate'
+ZESTIMATE = '{}:{}'.format(DEFAULT_NAME, NAME)
+
+ICON = 'mdi:home-variant'
+
+ATTR_LOCATION = 'location'
+ATTR_UPDATE = 'update'
+ATTR_AMOUNT = 'amount'
+ATTR_CHANGE = 'amount_change_30days'
+ATTR_CURRENCY = 'amount_currency'
+ATTR_LAST_UPDATED = 'amount_last_updated'
+ATTR_VAL_HI = 'valuation_range_high'
+ATTR_VAL_LOW = 'valuation_range_low'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_ZPID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Zestimate sensor."""
+    import xmltodict
+
+    name = config.get(CONF_NAME)
+    properties = config[CONF_ZPID]
+    params = {'zws-id': config[CONF_API_KEY]}
+
+    sensors = []
+    for zpid in properties:
+        params['zpid'] = zpid
+        try:
+            response = requests.get(_RESOURCE, params=params, timeout=5)
```

Comment:
Why do you do this? This looks like you do the first update before you create the `ZestimateData` object. But you don't use any of the results to construct the entity?

Please don't do this. Setting up the platform should be fast, i.e., should do no I/O if not absolutely necessary. The first update should happen only when your `update` method is called for the first time.